provider/core/src/datagen/mod.rs

```diff
@@ -21,8 +21,30 @@ pub use payload::{ExportBox, ExportMarker};
 
 use crate::prelude::*;
 
+/// The type of fallback that the data was generated for. Data size can be reduced as
+/// long as the data consumers know how this was done.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum FallbackMode {
+    /// No fallback
+    None,
+    /// Full fallback
+    Full,
```

Comment:
Nit: These settings names deserve bikeshedding. I would suggest

```suggestion
    /// Fallback will be performed at runtime, such that data can be smaller
    RuntimeFullFallback,
    /// The selected locales should be fully resolved at build time
    PreComputedFallback,
```

Reply:
`None`, i.e. the current behaviour is different to `PreComputedFallback` though. Currently if a user requests a locale and there's no data for it, we won't include fallback/root data. Do you want to get rid of this behaviour?